lib/Doctrine/ORM/Persisters/Entity/JoinedSubclassPersister.php

```diff
@@ -282,7 +283,11 @@ public function delete($entity)
             $rootTable = $this->quoteStrategy->getTableName($rootClass, $this->platform);
             $rootTypes = $this->getClassIdentifiersTypes($rootClass);
 
-            return (bool) $this->conn->delete($rootTable, $id, $rootTypes);
+            // Fix for bug GH-8229 (id column from parent class renamed in child class):
+            // Use the correct name for the id column as named in the root class.
+            $rootId = array_combine($rootClass->getIdentifierColumnNames(), $identifier);
+
+            return (bool) $this->conn->delete($rootTable, $rootId, $rootTypes);
```

Comment:
Why similar line at L297 still uses $id, is that correct?

Reply:
Yes, that's correct. The variable naming is a bit confusing, because it's also called $rootTable, although it isn't the root table, it's the current one (see comment "starting from this class' table up to the root table"), so $id is correct in this case. I didn't want to unnecessarily change existing code, but could rename it of course, to make it clearer.